docs/navigation-options-resolution.md

```diff
@@ -312,3 +312,31 @@ const AppNavigator = createSwitchNavigator({
   Home: HomeStack,
 });
 ```
+
+# A stack has more than one screen and you want to specify the transition mode for each screen explicitly
+
+We can’t set the `StackNavigatorConfig`’s `mode` dynamically. Instead we are going to use a custom `transitionConfig` to render the specfific transition we want - modal or card - on a screen by screen basis.
+
+```js
+  import { createStackNavigator, StackViewTransitionConfigs } from "react-navigation";
+
+  /* The screens you add to IOS_MODAL_ROUTES will have the modal transition.  */
+  const IOS_MODAL_ROUTES = ["OptionsScreen"];
+
+  let dynamicModalTransition = (transitionProps, prevTransitionProps) => {
```

Comment:
we should indicate what the third param to `defaultTransitionConfig` is here.

also, I don't think this doc is the right place for this because we're not configuring navigation options. I'd add this to the stack navigator reference doc

Reply:
So you mean by explicitly writing: "`isModal = IOS_MODAL_ROUTES.some(` ..." ?